### PR TITLE
Introduction to `FOLDER_TREE_LENGTH` config parameter

### LINKED ISSUE #1: Introduce `FOLDER_TREE_LENGTH` config parameter on `scrapy_folder_tree.FilesHashTreePipeline`
Given this filename: `abcdefg.png` and `FOLDER_TREE_LENGTH=2`

Currently, when using the `scrapy_folder_tree.FilesHashTreePipeline` pipeline, the file path will look like `a/b/abcdefg.png`.

An extra config value (`FOLDER_TREE_LENGTH`) should be added similar to `FOLDER_TREE_LENGTH` in order to allow the user to specify the folder length. For example, if `FOLDER_TREE_LENGTH` is equal to `2`, the file path should look like `ab/cd/abcdefg.png` etc.

In order to support this functionality, the extra config should be added [here](https://github.com/sp1thas/scrapy-folder-tree/blob/master/src/scrapy_folder_tree/config.py) and the filename should be split, based on the `FOLDER_TREE_LENGTH` [here](https://github.com/sp1thas/scrapy-folder-tree/blob/master/src/scrapy_folder_tree/trees/hash.py#L11).

### PR DESCRIPTION
**Pull request by HemantSachdeva**
Resolves #1.

Signed-off-by: Hemant Sachdeva <hemant.evolver@gmail.com>

### CODE REVIEW

File: src/scrapy_folder_tree/trees/hash.py [hunk 1]
```diff
@@ -4,10 +4,10 @@
...
-    def __init__(self, depth: int) -> None:
+    def __init__(self, depth: int, length: int) -> None:
         self.DEPTH = depth
+        self.LENGTH = length
 
     def build_path(self, filepath) -> str:
         filename = os.path.basename(filepath)
-        dir_name = os.path.dirname(filepath)
-        return os.path.join(dir_name, *[_ for _ in filename[: self.DEPTH]], filename)
+        return os.path.join(*[filename[i:i + self.LENGTH] for i in range(0, self.DEPTH * self.LENGTH, self.LENGTH)], filename)
```
Comment (sp1thas) [inline src/scrapy_folder_tree/trees/hash.py:13]: Pretty clean implementation @HemantSachdeva . Btw, `dirname` should remain as is in order to store the files in the necessary folder (regardless of the extra folder structure)
```suggestion
        dir_name = os.path.dirname(filepath)
        return os.path.join(dir_name, *[filename[i:i + self.LENGTH] for i in range(0, self.DEPTH * self.LENGTH, self.LENGTH)], filename)
```

**Review by sp1thas — CHANGES_REQUESTED** [review]

File: src/scrapy_folder_tree/trees/hash.py (hunk 1 above)
Comment (HemantSachdeva) [reply src/scrapy_folder_tree/trees/hash.py:13]: Sure  I'll re-add `dir_name`.
I'll also update the docs as #5.

**Review by sp1thas — APPROVED** [review]
LGTM! Thanks for your contribution @HemantSachdeva

**Review by sp1thas — CHANGES_REQUESTED** [review]
One more thing @HemantSachdeva apply black formatting [run](https://github.com/sp1thas/scrapy-folder-tree/actions/runs/3307370465/jobs/5459302585)

**Comment by HemantSachdeva** [comment]
Hope we are good to go now.
It was a nice experience for me :)

**Comment by HemantSachdeva** [comment]
Oh man [this](https://github.com/sp1thas/scrapy-folder-tree/actions/runs/3307780844/jobs/5459993613#step:7:9) black 😂🫡

**Pull request merged**